### PR TITLE
Fix method header memory unsafety/leaks

### PR DESCRIPTION
**Pull request by alexanderkyte**

### CODE REVIEW

File: mono/metadata/class-internals.h [hunk 1]
```diff
@@ -1044,8 +1044,10 @@ mono_method_inflated_lookup (MonoMethodInflated* method, gboolean cache);
 MONO_API MonoMethodSignature *
 mono_metadata_get_inflated_signature (MonoMethodSignature *sig, MonoGenericContext *context);
 
-MonoType*
-mono_class_inflate_generic_type_with_mempool (MonoImage *image, MonoType *type, MonoGenericContext *context, MonoError *error);
+MonoType *
+mono_class_inflate_generic_type_with_mempool (MonoImage *image, MonoType *type,
+																							MonoGenericContext *context, gboolean allow_sharing,
```
Comment (Therzok) [inline mono/metadata/class-internals.h]: ETOO_MANY_TABS.

File: mono/metadata/class.c [hunk 2]
```diff
@@ -803,28 +803,34 @@ mono_class_get_generic_class (MonoClass *klass)
  * modified by the caller, and it should be freed using mono_metadata_free_type ().
  */
 MonoType*
-mono_class_inflate_generic_type_with_mempool (MonoImage *image, MonoType *type, MonoGenericContext *context, MonoError *error)
+mono_class_inflate_generic_type_with_mempool (MonoImage *image, MonoType *type,
+																							MonoGenericContext *context, gboolean allow_sharing,
```
Comment (Therzok) [inline mono/metadata/class.c]: ETOO_MANY_TABS here too.

File: mono/metadata/class.c [hunk 3]
```diff
@@ -803,28 +803,34 @@ mono_class_get_generic_class (MonoClass *klass)
...
 		return NULL;
+	} else if (inflated) {
+		mono_stats.inflated_type_count++;
+		return inflated;
+	}
 
-	if (!inflated) {
-		MonoType *shared = mono_metadata_get_shared_type (type);
+	if (allow_sharing)
+		 shared = mono_metadata_get_shared_type (type);
```
Comment (Therzok) [inline mono/metadata/class.c]: Somebody else's gonna spot this so: EMIXED_LEADING_SPACE.

**Comment by lewurm** [comment]
is there still interest to merge this or should it be closed?

**Comment by kumpera** [comment]
This was fixed in a different way

**Pull request closed without merging**